llama_index/vector_stores/astra.py

```diff
@@ -0,0 +1,183 @@
+"""Astra DB Vector store index.
+
+An index based on a DB table with vector search capabilities,
+powered by the astrapy library
+
+"""
+import logging
+from typing import Any, List, Optional, cast
+
+from llama_index.schema import BaseNode, MetadataMode
+from llama_index.vector_stores.types import (
+    VectorStore,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+from llama_index.vector_stores.utils import (
+    metadata_dict_to_node,
+    node_to_metadata_dict,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class AstraDBVectorStore(VectorStore):
+    """Astra DB Vector Store.
+
+    An abstraction of a Astra table with
+    vector-similarity-search. Documents, and their embeddings, are stored
+    in an Astra table and a vector-capable index is used for searches.
+    The table does not need to exist beforehand: if necessary it will
+    be created behind the scenes.
+
+    All Astra operations are done through the astrapy library.
+
+    Args:
+        collection_name (str): collection name to use. If not existing, it will be created.
+        token (str): The Astra DB Application Token to use.
+        api_endpoint (str): The Astra DB JSON API endpoint for your database.
+        embedding_dimension (int): length of the embedding vectors in use.
+        namespace (Optional[str]): The namespace to use. If not provided, 'default_keyspace'
+        ttl_seconds (Optional[int]): expiration time for inserted entries.
+            Default is no expiration.
+
+    """
+
+    stores_text: bool = True
+    flat_metadata: bool = True
+
+    def __init__(
+        self,
+        *,
+        collection_name: str,
+        token: str,
+        api_endpoint: str,
+        embedding_dimension: int,
+        namespace: Optional[str] = None,
+        ttl_seconds: Optional[int] = None,
+    ) -> None:
+        import_err_msg = "`astrapy` package not found, please run `pip install astrapy`"
+
+        # Try to import astrapy for use
+        try:
+            from astrapy.db import AstraDB, AstraDBCollection
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        # Set all the required class parameters
+        self._embedding_dimension = embedding_dimension
+        self._ttl_seconds = ttl_seconds
+
+        _logger.debug("Creating the Astra DB table")
+
+        # Build the Astra DB object
+        self.astra_db = AstraDB(
```

Comment:
very nit: can we make these "private" `_astra_db`. I think we've been generally trying to adopt this standard throughout all of our vector store integrations (though some have slipped thru the cracks it seems)

Reply:
Done! Thanks!